### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by Sutanshu**
Bunch of comments, and individually liked
Screen Shot 2022-04-03 at 1 47 35 PM (2)

Adding a comment, will make this prettier in the next design PR
Screen Shot 2022-04-03 at 1 47 55 PM (2)

Added comment, also fixed navBar css
Screen Shot 2022-04-03 at 1 50 26 PM (2)

### CODE REVIEW

File: src/app/src/App.tsx [hunk 1]
```diff
@@ -34,22 +90,32 @@ function App() {
...
+              <>
+                <Mainpage currentUser={currentUser} />
+              </>
             ) : (
               <Homepage setCurrentUser={setCurrentUser} />
             )
           }
         />
+        <Route path="/profile/:id" element={<Profile currentUser={currentUser} />} />
+        <Route path="/authors/:id/posts/:id" element={<Comments currentUser={currentUser} />} />
```
Comment (osmani2) [inline src/app/src/App.tsx]: This looks like the backend route. We wanted the frontend route to just be `path="/post/:postId"` where you can see the post and all it's comments.
UPDATE: route to use it `/profile/:authorId/post/:postId`

File: src/app/src/components/UserPost.tsx [hunk 2]
```diff
@@ -193,34 +196,40 @@ const UserPost: React.FC<postItem> = (props?) => {
...
   //Navigate to user's profile from userpost
-  const HandleNavigation = () => { 
-    navigate(`/profile/${props?.postAuthor?.id.split('/').pop()}`) 
+  const HandleNavigation = () => {
+    navigate(`/profile/${props?.postAuthor?.id.split("/").pop()}`);
+  };
+
+  const navigateToComments = () => {
+    navigate(
+      `/authors/${props?.postAuthor?.id.split("/").pop()}/posts/${props?.post?.id}/comments`,
```
Comment (osmani2) [inline src/app/src/components/UserPost.tsx]: Same issue here.
This looks like the backend route. We wanted the frontend route to just be `path="/post/:postId"` where you can see the post and all it's comments.
UPDATE: route to use it `/profile/:authorId/post/:postId`

File: src/app/src/components/UserPost.tsx [hunk 3]
```diff
@@ -270,30 +279,34 @@ const UserPost: React.FC<postItem> = (props?) => {
...
+                      color: "red",
+                      "&:hover": {
+                        cursor: "pointer",
                       },
                     }}
                   />
                 )}
               </LikesContainer>
-              <CommentsContainer>{props?.post?.count} Comments</CommentsContainer>
+              <CommentsContainer onClick={navigateToComments}>
```
Comment (osmani2) [inline src/app/src/components/UserPost.tsx:305]: Can we change the cursor style to a pointer when hovering

File: src/app/src/pages/Mainpage.tsx [hunk 4]
```diff
@@ -85,9 +63,6 @@ export default function Mainpage({ currentUser }: Props) {
 
   return (
     <MainPageContainer>
-      <NavBarContainer>
```
Comment (osmani2) [inline src/app/src/pages/Mainpage.tsx:88]: Can we do this for the other pages, or were you planning on that in your design PR?
Screen Shot 2022-04-03 at 2 24 13 PM

**Review by osmani2 — CHANGES_REQUESTED** [review]

File: src/app/src/pages/Comments.tsx [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+  "&:hover": {
+    backgroundColor: "#b5b5b5",
+  },
+}));
+const formStyle = { m: 1, minWidth: 120, width: "40%", mt: 2 };
+
+export default function Comments({ currentUser }: Props): JSX.Element {
+  const url = new URL(window.location.href).pathname;
+  const postID = url.split("/")[4];
+  const authorID = url.split("/")[2];
```
Comment (kylemclean) [inline src/app/src/pages/Comments.tsx]: Use `useParams()` instead.
https://reactrouter.com/docs/en/v6/api#useparams

**Review by kylemclean — CHANGES_REQUESTED** [review]

File: src/app/src/App.tsx (hunk 1 above)
Comment (osmani2) [reply src/app/src/App.tsx]: @Sutanshu I think I was unclear with this, sorry. I DON'T want this to look like the backend route. It is the frontend route and needs to be `/profile/:authorId/post/:postId`

File: src/app/src/components/UserPost.tsx (hunk 2 above)
Comment (osmani2) [reply src/app/src/components/UserPost.tsx]: @Sutanshu Same here again, should be of the form `/profile/${props?.postAuthor?.id.split("/").pop()}/post/${props?.post?.id}` same as App.tsx

File: src/app/src/pages/Comments.tsx [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+  }, [authorID, postID, commentsChanged]);
+
+  const comment = {
+    comment: content,
+    contentType: type,
+  };
+  const createComment = () => {
+    if (authorID && postID) {
+      api.authors
+        .withId(authorID)
```
Comment (osmani2) [inline src/app/src/pages/Comments.tsx:129]: We'd want the currentUser to create the comment. Not the authorID one, that's who'se post we're commenting on.

File: src/app/src/pages/Comments.tsx [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+                "&:hover": { background: "#18E78F" },
+              }}
+            >
+              <CheckIcon onClick={createComment} />
+            </Fab>
+          </EditContainer>
+        </Backdrop>
+      ) : null}
+      {post ? (
+        <UserPost post={post} likes={5} currentUser={currentUser} postAuthor={currentUser} />
```
Comment (osmani2) [inline src/app/src/pages/Comments.tsx]: We want the authorID author as postAuthor (use authorID to get author), we're not always commenting on our own posts.

**Review by osmani2 — CHANGES_REQUESTED** [review]

**Review by osmani2 — APPROVED** [review]

**Pull request merged**